src/Symfony/Component/HttpClient/CurlHttpClient.php

```diff
@@ -274,6 +274,11 @@ public function request(string $method, string $url, array $options = []): Respo
             $curlopts[CURLOPT_TIMEOUT_MS] = 1000 * $options['max_duration'];
         }
 
+        if (isset($options['extra']['curl']) && \is_array($options['extra']['curl'])) {
+            $curlopts += $options['extra']['curl'];
+            $this->logger && $this->logger->debug(sprintf('Default curl options has been overriden.'));
```

Comment:
if we dont allow to override computed options, perhaps log those options that are ignored.

Reply:
we even could throw an InvalidArgumentException

Reply:
how about taking an array of all curlopts that can be defined by using $options keys, 
and break if any of them will try to be overriden by extra.curl?

e.g: 
````
$fixedCurlopts = [
	CURLOPT_TIMEOUT_MS => 'max_duration',
	CURLOPT_UNIX_SOCKET_PATH => 'bindto',
	CURLOPT_INTERFACE => 'bindto'
	
	//etc.
]

$overridenOpts = array_keys($options['extra']['curl']);

foreach($overridenOpts as $option) {
	if(isset($fixedCurlopts[$option]) {
		throw new \InvalidArgumentExceptions(sprintf('Could not override %s curlopt, please use %s option', $option, $fixedCurlopts[$option]);
	}
}

//check CURLOPT_ for methods etc.
//if any exception thrown at this moment, merge with $curlopts
````

Forces to use default options and allows only few options to be set manually.